File: algorithms/cpp/343.cpp
#include <iostream>
using namespace std;

class Solution {
public:
    int integerBreak(int n) {
        switch (n) {
            case 2: return 1;
            case 3: return 2;
            case 4: return 4;
            case 5: return 6;
            case 6: return 9;
        }
        return 3 * integerBreak(n-3);
    }
};

int main() {
    return 0;
}

File: algorithms/python3/343.py
class Solution:
    def integerBreak(self, n):
        """
        :type n: int
        :rtype: int
        """
        if n == 2:
            return 1
        elif n == 3:
            return 2
        elif n % 3 == 0:
            return 3 ** (n//3)
        elif n % 3 == 1:
            return 3 ** ((n-4)//3) * 4
        else:
            return 3 ** (n//3) * 2
